meta-cli/src/typegraph/postprocess.rs

```diff
@@ -99,23 +103,50 @@ pub mod deno_rt {
         }
     }
 
-    fn reformat_materializer_script(mat: &mut Materializer) -> Result<()> {
-        match mat.name.as_str() {
-            "function" => {
-                let mut mat_data: FunctionMatData = object_from_map(std::mem::take(&mut mat.data))?;
-                // TODO check variable `_my_lambda` exists and is a function expression/lambda
-                mat_data.script = transform_script(mat_data.script)?;
-                mat.data = map_from_object(mat_data)?;
-            }
-            "module" => {
-                let mut mat_data: ModuleMatData = object_from_map(std::mem::take(&mut mat.data))?;
-                if !mat_data.code.starts_with("file:") {
-                    // TODO check imported functions exist
-                    mat_data.code = transform_module(mat_data.code)?;
+    fn compress_and_encode(main_path: &Path, tg_path: &Option<PathBuf>) -> String {
+        // Note: main_path and tg_path are all absolute
+        // tg_root/tg.py
+        // tg_root/scripts/deno/* <= script location
+        let base_rel_path = match tg_path {
+            Some(path) => {
+                let tg_folder = path.parent();
+                if let Some(tg_folder) = tg_folder {
+                    tg_folder.display().to_string()
+                } else {
+                    "".to_string()
                 }
-                mat.data = map_from_object(mat_data)?;
             }
-            _ => {}
+            None => "".to_string(),
+        };
+
+        let dir_walker = WalkBuilder::new(&base_rel_path)
+            .standard_filters(true)
+            .sort_by_file_path(|a, b| a.cmp(b))
+            .build();
+        let suffix = Path::new("scripts/deno");
+        let enc_content = match archive_entries(dir_walker, Some(suffix)).unwrap() {
+            Some(b64) => b64,
+            None => "".to_string(),
+        };
+        // main_path: /abs/path/to/my_typegraphs/scripts/deno/ts/dep/main.ts
+        // base_path: relative/my_typegraphs (based on cwd)
+        let file = main_path
+            .display()
+            .to_string()
+            .split(&base_rel_path)
+            .last() // TODO: what if user use a subpath == base_rel_path ?
+            .unwrap()
+            .to_owned();
+
+        format!("file:{};base64:{}", file, enc_content)
+    }
+
+    fn reformat_materializer_script(mat: &mut Materializer) -> Result<()> {
+        if mat.name.as_str() == "function" {
+            let mut mat_data: FunctionMatData = object_from_map(std::mem::take(&mut mat.data))?;
+            // TODO check variable `_my_lambda` exists and is a function expression/lambda
```

Comment:
todo before merge?

Reply:
This comment was there before, I just removed the unused code then replaced the `match` to a `if`. Good to have indeed, but I assume it wouldn't be that trivial to do properly.